packages/components/src/components/Cell/NavigationCell.tsx

```diff
@@ -29,55 +46,97 @@ export type NavigationCellProps = {
      * Name of the icon to use as a left adornment.
      */
     iconName?: IconName;
-} & Omit<CellProps, "leftAdornment" | "rightAdornment" | "onPress">;
+} & NavigationCellOptions &
+    Omit<CellProps, "leftAdornment" | "rightAdornment" | "onPress">;
 
 export const NavigationCell = ({
     title,
     onPress,
     disabled = false,
     description,
     iconName,
+    additionalTitlesRight,
+    additionalTitlesUnder,
     ...cellProps
 }: NavigationCellProps) => {
     const styles = useStyles(themeStyles);
 
-    const IconAdornment = () => (
+    const renderAdornmentIcon = (iconName: IconName) => (
         <View style={styles.adornmentContainer}>
-            <Icon name={iconName ?? "dots-square"} color={disabled ? "textDisabled" : undefined} />
+            <Icon name={iconName} color={disabled ? "textDisabled" : undefined} />
         </View>
     );
 
-    const DisclosureAdornment = () => (
-        <View style={styles.adornmentContainer}>
-            <Icon name="chevron-right" color={disabled ? "textDisabled" : undefined} />
-        </View>
-    );
+    const renderAdditionalTitlesRight = () =>
+        additionalTitlesRight?.map((text, index) => (
+            <Typography
+                key={index}
+                group="cell"
+                variant="title"
+                numberOfLines={1}
+                color="textTertiary"
+                style={[additionalTitlesRight && { flex: 1 }]}
```

Comment:
is this conditional on the styling needed? It seems `renderAdditionalTitlesRight` already is conditionally called for.